tests/models/test_default_evaluator.py

```diff
@@ -684,3 +693,221 @@ def example_custom_metric(eval_df, given_metrics):
 
     assert "positive_count" in result.metrics
     assert np.isclose(result.metrics["positive_count"], np.sum(y_pred), rtol=1e-3)
+
+
+def test_load_custom_metric_artifact_returns_given_evaluation_artifact_as_is():
```

Comment:
Rather than testing a private / internal function, can we instead structure these as tests of the `mlflow.evaluate()` public API with `evaluators="default"`?

Reply:
Yes, I am doing that refactor right now. Will update this PR soon :)

Reply:
Yes, it was intentional that nowhere else calls `_load_custom_metric_artifact` in this PR. Because my original intention was to have a follow up PR that calls this function and to connect it with `mlflow.evaluate`. But now it seems like it would make more sense to have that follow up PR combined with this.